File: Hello-libs/app/src/main/java/com/ccooy/hellolibs/MainActivity.java
package com.ccooy.hellolibs;

import android.os.Bundle;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TextView tv = new TextView(this);
        tv.setText(stringFromJNI());
        setContentView(tv);
    }

    public native String stringFromJNI();

    static {
        System.loadLibrary("hello-libs");
    }

}

File: Hello-cdep/app/src/main/java/com/ccooy/hellocdep/MainActivity.kt
package com.ccooy.hellocdep

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var tv = TextView(this)
        tv.text = stringFromJNI()
        setContentView(tv)
    }

    external fun stringFromJNI(): String

    companion object {
        init {
            System.loadLibrary("hello-cdep")
        }
    }
}
